src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -1044,6 +1051,10 @@ citus_drain_node(PG_FUNCTION_ARGS)
 	};
 
 	char *nodeName = text_to_cstring(nodeNameText);
+	if (nodePort != 0 && (nodeName != NULL && strlen(nodeName) > 0))
```

Comment:
I don't think this check is necessary

Reply:
How can we understand if a worker passed as a parameter without this check?

Reply:
This function should only be passed a valid worker node. Probably master_set_node_property checks that already. But since that's run in a separate transaction it's best not to depend on that TBH. 

So, I think you're right that we need some check. But I don't think we need this one for the following reason:
1. nodeName will never be NULL, because of `PG_ENSURE_ARGNOTNULL(0, "nodename");` above
2. FindWorkerNode handles all other cases already

I think removing this if and replacing `FindWorkerNode` with `FindWorkerNodeOrError` should be enough.